include/deal.II/fe/fe_tools.templates.h

```diff
@@ -103,22 +100,78 @@ namespace FETools
             break;
           }
 
+      bool initialized = false;
+
       for (unsigned int i = 0; i < fes.size(); i++)
         if (multiplicities[i] > 0)
           {
-            // TODO: the implementation makes the assumption that all faces have
-            // the same number of dofs -> don't construct DPO but
-            // PrecomputedFiniteElementData
-            AssertDimension(fes[i]->n_unique_quads(), 1);
-
-            multiplied_dofs_per_vertex +=
-              fes[i]->n_dofs_per_vertex() * multiplicities[i];
-            multiplied_dofs_per_line +=
-              fes[i]->n_dofs_per_line() * multiplicities[i];
-            multiplied_dofs_per_quad +=
-              fes[i]->n_dofs_per_quad(0) * multiplicities[i];
-            multiplied_dofs_per_hex +=
-              fes[i]->n_dofs_per_hex() * multiplicities[i];
+            if (initialized == false)
```

Comment:
If I read this right, what you are doing here is initialize the dpo object the first time you find an element with `multiplicity>0`. I would much rather you do this before the current `for` loop, as in
```
  for (i=...)
    if (multiplicity>0)
      {
         ...do it here...
         break;
      }
```
I think that would be easier to read, and avoid the `initialized` variable altogether.